spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UnicastSendingMessageHandler.java

```diff
@@ -99,6 +109,35 @@ public UnicastSendingMessageHandler(String host, int port) {
 		super(host, port);
 		this.mapper.setLengthCheck(false);
 		this.mapper.setAcknowledge(false);
+		this.socketExpression = null;
+	}
+
+	/**
+	 * Basic constructor using external datagram socket for sending outgoing data.
+	 * @param socketExpression socket expression string describing external socket
+	 * @since 4.3
+	 */
+	public UnicastSendingMessageHandler(String socketExpression) {
+		super("", 0);
+		Assert.hasText(socketExpression);
+		Assert.state(!this.acknowledge, "socketExpression cannot be used together with acknowledge");
```

Comment:
This must not be here, because `acknowledge` is populated from the setter. So that one assert in the `onInit()` is enough. It is called from the `afterPropertiesSet()` by Spring container.
So, we are safe here anyway.